File: cs50x/week6/sentimental-credit_f/credit.py
# TODO
def main():
    while True:
        try:
            card = input("Card number: ")
            return calculate(card)
        except ValueError:
            print("NOT A NUMBER")


def calculate(n):
    sum = 0
    c = int(n)

    for _ in range(len(n)):
        num = int(((c % 100)) / 10) * 2
        sum += ((num % 10) + (int((num % 100) / 10)) + (c % 10))
        c = int(c / 100)

    if sum % 10 == 0 and len(n) in [13, 15, 16]:
        if 34 <= int(n[:2]) <= 37:
            print("AMEX")
        elif 51 <= int(n[:2]) <= 55:
            print("MASTERCARD")
        elif int(n[:1]) == 4:
            print("VISA")
        else:
            print("INVALID")
    else:
        print("INVALID")


if __name__ == "__main__":
    main()

File: cs50x/week1/credit_f/credit.c
#include <cs50.h>
#include <stdio.h>

int calculator(long c_num);

int main(void)
{
   long c_num;
   do
   {
        c_num = get_long ("Card number: ");
   }
   while (c_num < 0);

   calculator(c_num);
}



int calculator(long c_num)
{
   int sum = 0;
   long i_num = c_num;

   for(int i = 16; i > 0; i--)
   {
      int num_2 = ((i_num % 100) / 10) * 2;

      sum += i_num % 10;

      sum += (num_2 % 10) + ((num_2 % 100) / 10);

      i_num /= 100;
   }

   if ((c_num / 10000000000000) >= 34 && (c_num / 10000000000000) < 38 && sum % 10 == 0 )
   {
      printf("AE\n");
   }

   else if ((c_num / 100000000000) >= 40 && (c_num / 100000000000) < 50 && sum % 10 == 0)
   {
      printf("Visa\n");
   }

   else if ((c_num / 10000000000000) >= 40 && (c_num / 100000000000000) < 50 && sum % 10 == 0)
   {
      printf("Visa\n");
   }

   else if ((c_num / 100000000000000) >= 51 && (c_num / 100000000000000) < 55 && sum % 10 == 0)
   {
      printf("MC\n");
   }
   else
   {
      printf("Invalid\n");
   }
   return 0;
}

